### PR TITLE
display license info for helm app installations

### PR DESCRIPTION
**Pull request by stefanrepl**
#### What this PR does / why we need it:
Get license from replicated app endpoint for helm installations.  Then display the license data in kots UI (refer to screenshot below)
#### Which issue(s) this PR fixes:

Fixes #
https://app.shortcut.com/replicated/story/51369/show-updated-license-information-in-the-admin-console
#### Does this PR introduce a user-facing change?

```release-note
     Display license information in UI for Helm application installs
```

#### Does this PR require documentation?

Screen Shot 2022-07-07 at 2 09 02 PM
Screen Shot 2022-07-07 at 2 09 09 PM

### CODE REVIEW

File: pkg/handlers/license.go [hunk 1]
```diff
@@ -173,20 +176,76 @@ func (h *Handler) GetLicense(w http.ResponseWriter, r *http.Request) {
...
+		defer resp.Body.Close()
+		responseBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			getLicenseResponse.Error = "failed to read response body"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+
+		license, err = pull.ParseLicenseFromBytes(responseBody)
```
Comment (sgalsaleh) [inline pkg/handlers/license.go:226]: i'd suggest using `kotsutil`
```suggestion
		license, err = kotsuitl.LoadLicenseFromBytes(responseBody)
```

File: pkg/handlers/license.go [hunk 2]
```diff
@@ -173,20 +176,76 @@ func (h *Handler) GetLicense(w http.ResponseWriter, r *http.Request) {
...
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			getLicenseResponse.Error = "failed to perform http request"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+
+		defer resp.Body.Close()
+		responseBody, err := ioutil.ReadAll(resp.Body)
```
Comment (sgalsaleh) [inline pkg/handlers/license.go:218]: should check for status code first and fail if it's not the expected code

File: pkg/handlers/license.go [hunk 3]
```diff
@@ -173,20 +176,76 @@ func (h *Handler) GetLicense(w http.ResponseWriter, r *http.Request) {
...
-		getLicenseResponse.Error = "failed to get app from slug"
-		logger.Error(errors.Wrap(err, getLicenseResponse.Error))
-		JSON(w, http.StatusInternalServerError, getLicenseResponse)
-		return
-	}
+	license := new(kotsv1beta1.License)
+	foundApp := new(apptypes.App)
+
+	isHelmManaged := os.Getenv("IS_HELM_MANAGED")
+	if isHelmManaged == "true" {
```
Comment (sgalsaleh) [inline pkg/handlers/license.go:183]: how about moving this block to a separate function so that it's a bit cleaner?

**Review by rianfowler — APPROVED** [review]
FE approved

File: pkg/handlers/helm.go [hunk 4]
```diff
@@ -72,3 +77,45 @@ func (h *Handler) GetAppValuesFile(w http.ResponseWriter, r *http.Request) {
...
+	foundApp := &apptypes.App{ID: chartApp.ID, Slug: chartApp.Slug, Name: chartApp.Name}
+	apiEndpoint := os.Getenv("REPLICATED_API_ENDPOINT")
+
+	// get license
+	req, err := util.NewRequest(http.MethodGet, fmt.Sprintf("%s/license", apiEndpoint), nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	var licId string
+	if appCache[chartName].Values["replicated"] != nil && appCache[chartName].Values["replicated"].(map[string]interface{})["license_id"] != nil {
```
Comment (divolgin) [inline pkg/handlers/helm.go:93]: This needs some error checking.  `appCache[chartName]` can be nil

File: pkg/handlers/license.go [hunk 5]
```diff
@@ -173,20 +174,34 @@ func (h *Handler) GetLicense(w http.ResponseWriter, r *http.Request) {
...
+		license, foundApp, err = getLicenseForHelmApp(appSlug)
+		if err != nil {
+			getLicenseResponse.Error = "failed to get license for helm app"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+	} else {
+		foundApp, err := store.GetStore().GetAppFromSlug(appSlug)
```
Comment (sgalsaleh) [inline pkg/handlers/license.go:190]: this doesn't override the top-level `foundApp` variable

**Pull request merged**